js/main.js

```diff
@@ -290,32 +315,69 @@ function renderDetail(result) {
 
 function handleDetail(event) {
   var $dataView = event.target.getAttribute('data-view');
+
   if (event.target.nodeName === 'BUTTON' && $dataView !== '') {
     var $previousRender = document.querySelectorAll('#detail-page-render');
-
     for (var d = 0; d < $previousRender.length; d++) {
       $previousRender[d].remove();
     }
     viewSwap($dataView);
   }
 
-  var $resultId = event.target.getAttribute('result-id');
-
-  var render = renderDetail(data.results[$resultId]);
-  $detailPageContainer.appendChild(render);
-  data.detail = data.results[$resultId];
+  var $resultId = event.target.getAttribute('data-id');
 
+  if (data.searchPageView === 'Bookmarks') {
+    var renderBookmarks = renderDetail(data.bookmarks[$resultId]);
+    $detailPageContainer.appendChild(renderBookmarks);
+    data.detail = data.bookmarks[$resultId];
+    data.detailId = parseInt($resultId);
+  } else {
+    var render = renderDetail(data.results[$resultId]);
+    $detailPageContainer.appendChild(render);
+    data.detail = data.results[$resultId];
+    data.detailId = parseInt($resultId);
+  }
 }
 
-function handleAdd(event) {
-  var $class = event.target.getAttribute('class');
-  if (event.target.className === 'fas fa-plus margin-zero add-to-bookmarks' && $class.includes('add-to-bookmarks')) {
-    var $resultId = event.target.getAttribute('result-id');
+function handleAddAndRemove(event) {
+  if (event.target.nodeName === 'BUTTON' && event.target.textContent === 'Add to Bookmarks') {
+    $confirmMessage.textContent = 'Added to Bookmarks';
+    $bookmarkButtonModal.textContent = 'Go to Bookmarks';
+    $modal.className = 'modal-overlay';
+    data.bookmarks.push(data.results[parseInt(data.detailId)]);
+  } else if (event.target.nodeName === 'BUTTON' && event.target.textContent === 'Remove from Bookmarks') {
+    $modal.className = 'modal-overlay';
+    $confirmMessage.textContent = 'Remove from Bookmarks?';
+    $bookmarkButtonModal.textContent = 'Remove';
+  }
 
+  if (event.target.className === 'fas fa-plus margin-zero') {
+    var $resultId = event.target.getAttribute('data-id');
+    $confirmMessage.textContent = 'Added to Bookmark';
+    $bookmarkButtonModal.textContent = 'Go to Bookmarks';
     $modal.className = 'modal-overlay';
     event.target.className = 'fas fa-minus margin-zero';
-
     data.bookmarks.push(data.results[$resultId]);
+  } else if (event.target.className === 'fas fa-minus margin-zero') {
+    $resultId = event.target.getAttribute('data-id');
+    $modal.className = 'modal-overlay';
+    $confirmMessage.textContent = 'Remove from Bookmarks?';
+    $bookmarkButtonModal.textContent = 'Remove';
+  }
```

Comment:
Is this doing a similar job to the above conditional? It seems like it's trying to determine if an "add" or "remove" icon was licked, which is what the above is doing too, I believe. Can they be merged?